shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -144,6 +144,11 @@ - (instancetype)initWithEngine:(FlutterEngine*)engine
   self = [super initWithNibName:nibName bundle:nibBundle];
   if (self) {
     _viewOpaque = YES;
+    if (engine.viewController) {
+      FML_LOG(ERROR) << "Attempting to share a FlutterEngine across multiple "
```

Comment:
The supplied FlutterEngine << engine << is already used with FlutterViewController instance << engine.ViewController <<. One instance of the FlutterEngine can only be attached to one FlutterViewController at a time. Set FlutterEngine.viewController to nil before attaching it to another FlutterViewController. 

While on this topic, should we do this automatically in the FlutterViewController's dealloc too? Unsetting the engine's viewController property.

Reply:
> While on this topic, should we do this automatically in the FlutterViewController's dealloc too? Unsetting the engine's viewController property.

That already happens, niling out an engine's view controller with their view controller is dealloc'd.

Reply:
message updated.